helpers/chatHelper.py

```diff
@@ -266,462 +264,629 @@ async def partChannel(
-            # Make sure the channel exists
-            channel = await channelList.getChannel(to)
-            if channel is None:
-                raise exceptions.channelUnknownException()
-
-            # Make sure the channel is not in moderated mode
-            if channel["moderated"] and not osuToken.is_staff(userToken["privileges"]):
-                raise exceptions.channelModeratedException()
-
-            # Make sure we are in the channel
-            if to not in await osuToken.get_joined_channels(token_id):
-                # I'm too lazy to put and test the correct IRC error code here...
-                # but IRC is not strict at all so who cares
-                raise exceptions.userNotInChannelException()
-
-            # Make sure we have write permissions.
-
-            # premium requires premium
-            if (
-                to == "#premium"
-                and userToken["privileges"] & privileges.USER_PREMIUM == 0
-            ):
-                raise exceptions.channelNoPermissionsException()
 
-            # supporter requires supporter
-            if (
-                to == "#supporter"
-                and userToken["privileges"] & privileges.USER_DONOR == 0
-            ):
-                raise exceptions.channelNoPermissionsException()
+# TODO: add a functionality to handle sending to both IRC and bancho-service
+async def enqueue_private_message(
+    user_token: osuToken.Token,
+    recipient_token: osuToken.Token,
+    message: str,
+) -> None:
+    packet = serverPackets.sendMessage(
+        fro=user_token["username"],
+        to=recipient_token["username"],
+        message=message,
+        fro_id=user_token["user_id"],
+    )
+    await osuToken.enqueue(recipient_token["token_id"], packet)
+
+
+async def enqueue_public_message(
+    user_token: osuToken.Token,
+    channel_name: str,
+    client_channel_name: str,
+    message: str,
+    but: list[str] = [],
+) -> None:
+    packet = serverPackets.sendMessage(
+        fro=user_token["username"],
+        to=client_channel_name,
+        message=message,
+        fro_id=user_token["user_id"],
+    )
+
+    await streamList.broadcast(
+        f"chat/{channel_name}",
+        packet,
+        but=but,
+    )
+
+
+async def enqueue_public_limited_message(
+    user_token: osuToken.Token,
+    channel_name: str,
+    client_channel_name: str,
+    message: str,
+    send_to: list[str] = [],
+) -> None:
+    packet = serverPackets.sendMessage(
+        fro=user_token["username"],
+        to=client_channel_name,
+        message=message,
+        fro_id=user_token["user_id"],
+    )
+
+    await streamList.broadcast_limited(
+        f"chat/{channel_name}",
+        packet,
+        send_to,
+    )
+
+
+async def handle_private_bot_response(
+    user_token: osuToken.Token,
+    response: Optional[chatbot.CommandResponse],
+) -> None:
+    if response is None:
+        return
+
+    aika_token = await tokenList.getTokenFromUserID(CHATBOT_USER_ID)
```

Comment:
hmm how many places in the service are using "aika" in the var names? we should probably use "chatbot" uniformly, should be more timeless

Reply:
pretty nit though, can come later